group-coordinator/src/main/java/org/apache/kafka/coordinator/group/modern/ModernGroupMember.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.coordinator.group.modern;
+
+import org.apache.kafka.common.Uuid;
+import org.apache.kafka.coordinator.group.generated.ConsumerGroupCurrentMemberAssignmentValue;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Abstract member common for group members.
+ */
+public abstract class ModernGroupMember {
+
+    /**
+     * The member id.
+     */
+    protected String memberId;
+
+    /**
+     * The current member epoch.
+     */
+    protected int memberEpoch;
+
+    /**
+     * The previous member epoch.
+     */
+    protected int previousMemberEpoch;
+
+    /**
+     * The member state.
+     */
+    protected MemberState state;
+
+    /**
+     * The instance id provided by the member.
+     */
+    protected String instanceId;
```

Comment:
Do we suppose instance ids in share groups?

Reply:
We don't use in Share groups but the common in TargetAssignmentBuilder and elsewhere utilizes that. Hence to reuse the complete functionality it's been kept here.